CMakeLists.txt

```diff
@@ -491,11 +495,11 @@ if (GFLAGS_IS_SUBPROJECT)
   foreach (type IN ITEMS static shared)
     foreach (opts IN ITEMS "_nothreads" "")
       if (TARGET gflags${opts}_${type})
-        add_library (gflags ALIAS gflags${opts}_${type})
+        add_library (gflags::gflags ALIAS gflags${opts}_${type})
```

Comment:
I'm not sure, but I doubt you would use the `NAMESPACE` passed on to `install(EXPORT)` or `export` in the target name. Wouldn't this result in exported target names `gflags::gflags::<name>`? Also, the left hand of `ALIAS` does not use the `gflags::` prefix, while below you define the targets including that prefix.

Reply:
ALIAS can't be exported.
The point is when using `gflags` in my project Foo by using  `find_package(gflags)` or `add_subdirectory(gflags)` in both cases, I can use `target_link_library(foo gflags::gflags)`.

cf  https://gist.github.com/mbinna/c61dbb39bca0e4fb7d1f73b0d66a4fd1#using-a-library-defined-in-the-same-cmake-tree-should-look-the-same-as-using-an-external-library

Reply:
Yes, I understand the motivation. But I wonder if the exported target will be prefixed twice with `gflags::`? Because of the use of `NAMESPACE` when exporting targets that already have a `gflags::` prefix.

Reply:
Need to double check, but I don't think you export this target alias...

Reply:
I'm referring to the other targets, though. The ones the alias points to. I think also these were prefixed by `gflags::` as they should be for consistency reasons. But then the `add_library(ALIAS)` command would be incorrect I think. Will try it out before merging.

Reply:
Ok, I see now that `gflags::` prefix was only added to the ALIAS target, not the other targets. And then accordingly the template for the `gflags-config.cmake` file has been updated. Looks good, except that users that used the previous `gflags` alias will need to update their CMake files.

Given the (more or less) uniqueness of the `gflags` name already without the `gflags::` namespace prefix, do you have any objection to define both `gflags` and `gflags::gflags` aliases? This wouldn't break the build of dependent projects.

Reply:
define both `ALIAS` seems "OK". Not sure what to do with `gflags-config.cmake` should we duplicate all the targets ? with/without namespace ?

Reply:
Sorry, got the reply later/after merging. I'm not a fan of duplicating the targets. For the alias, that's only for when the library is used as subproject via `add_subdirectory` in which case it's fine to have `gflags` and `gflags::gflags` aliases.

I'll see how to best reduce impact for existing projects that use `find_package(gflags)`.